src/core/positioning/bluetoothreceiver.cpp

```diff
@@ -27,8 +27,10 @@ BluetoothReceiver::BluetoothReceiver( const QString &address, QObject *parent )
   , mLocalDevice( std::make_unique<QBluetoothLocalDevice>() )
   , mSocket( new QBluetoothSocket( QBluetoothServiceInfo::RfcommProtocol ) )
 {
-  connect( mSocket, &QBluetoothSocket::stateChanged, this, &BluetoothReceiver::setSocketState );
   connect( mSocket, qOverload<QBluetoothSocket::SocketError>( &QBluetoothSocket::errorOccurred ), this, &BluetoothReceiver::handleError );
+  connect( mSocket, &QBluetoothSocket::stateChanged, this, [=]( QBluetoothSocket::SocketState state ) {
+    setSocketState( static_cast<QAbstractSocket::SocketState>( state ) );
```

Comment:
I don't think we can blindly re-cast here, because QBluetoothSocket::SocketState::ClosingState's value is 5 and QAbstractSocket::ClosingState's value is 6.

We'll need to do a switch ( state ) to insure each bluetooth socket state matches the abstract socket equivalent.